rfc/halyard-lite.md

```diff
@@ -0,0 +1,475 @@
+insulates the existing Halyard from quick iteration on Halyard Lite. One disadvantage
+of this approach is that there will be a period of time where we need to maintain
+both the old and new Halyard.
+
+#### CLI vs. Daemon
+
+Halyard is currently implemented as a CLI and a daemon, greatly adding to its
+complexity.  This decision was made to allow support for clients other than the
+CLI, but to date no such implementations have been written and none are planned.
+The new tool will replace this with a CLI only, with no background daemon process.
+
+#### Language
+
+Halyard Lite will be written in Go, which has the following advantages:
+*   The primary initial consumer of Halyard Lite will be the Armory Operator,
+which is also written in Go and will be able to consume parts of Halyard Lite
+as a library
+*   Operators deploying to Kubernetes are usually much more familiar with Go than
+with Java, so this will likely lead to an easier path to contribution from the
+community
+*   Go is well-suited for writing simple command-line tools
+
+## Drawbacks
+
+This change will require users to change their workflow for deploying Spinnaker in
+order to adopt Halyard Lite. While the format of the Halyard config itself will not
+change, users will need to change the way they actually deploy Spinnaker from running
+`hal deploy apply` to using one of the above-described install pathways.
+
+## Prior Art and Alternatives
+
+ One alternative to the current approach of writing a new tool to replace Halyard
+ would be to refactor the existing Halyard to accomplish the same goals. That being
+ said, given the assessment above that the majority of Halyard's functionality is either
+ low-value or exists in other popular tools, the amount of code reuse that would be
+ possible by an in-place refactor is minimal. An in-place refactor would also add the
+ risk of breaking existing users during the implementation of this RFC.
+ 
+ Overall, one of the primary motivations of this RFC is to remove functionality
+ from Halyard that duplicates existing tools with larger market share, and instead
+ build a tool that integrates well with these existing tools. This has the benefit
+ of allowing users to install Spinnaker using tools that are familiar to them, and
+ removes the maintenance burden of Halyard's custom implementation of duplicative
+ tools.
+
+## Known Unknowns
+
+At this point, there are still a few details that are unknown and will be clarified
+during the implementation and testing phases:
+* The exact format of the output from the CLI tool. It will need to output the service
+configs as well as (possibly) some metadata about the Spinnaker deployment (ex: the versions
+of containers to pull) but the exact format will be determined in concert with early consumers.
+* The exact format of the Helm chart and Kustomize kustomization that will be provided to support
+installation to Spinnaker remains unspecified.
```

Comment:
Also: How do we correctly communicate that, yes, Spinnaker is k8s native via this work, without making non-k8s users feel abandoned or pressured to move onto k8s.

Reply:
One of my hopes with the stats collection is to be able to quantify these groups - how many deployments are going against which providers.

Reply:
I'm not sure I follow - are you implying that we're hoping k8s has significant enough adoption we don't need to address the rest?

Reply:
I've added a point to "Non-goals" to explicitly state that this document does not intend to deprecate non-Kubernetes install pathways (even as it focuses on making Kubernetes installs easier).